### PR TITLE
model: add cascade to all foreign keys

### PR DESCRIPTION
**Pull request by krzysztof**
Authored-by: David Caro <david@dcaro.es>
Co-authored-by: Krzysztof Nowak <k.nowak@cern.ch>
Signed-off-by: Krzysztof Nowak <k.nowak@cern.ch>

### CODE REVIEW

File: invenio_oauth2server/models.py [hunk 1]
```diff
@@ -162,7 +162,9 @@ class Client(db.Model):
         default=u'',
     )
 
-    user_id = db.Column(db.ForeignKey(User.id), nullable=True)
+    user_id = db.Column(
+        db.ForeignKey(User.id, ondelete='CASCADE'),
```
Comment (jmartinm) [inline invenio_oauth2server/models.py]: One thing we saw recently on INSPIRE when deleting many rows with ondelete='CASCADE' is that if the fields are not indexed, the operation can take forever. So might be better to already add `index=True`

Comment (krzysztof) [inline invenio_oauth2server/models.py]: @jmartinm Thanks for sharing. Fixed :)

File: examples/adminapp.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
```
Comment (lnielsen) [inline examples/adminapp.py]: Is the normal example app not enough?

File: invenio_oauth2server/admin.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        column_searchable_list = \
+        column_sortable_list = \
+        column_details_list = \
+        list_all
+
+
+oauth2server_clients_adminview = {
+    'model': Client,
+    'modelview': ClientView,
+    'category': _('OAuth2Server'),
```
Comment (lnielsen) [inline invenio_oauth2server/admin.py]: Should we put this under the User Management category?

**Pull request merged**